### PR TITLE
Add invoice transaction create [WIP]

### PR DESCRIPTION
**Pull request by mikereinmiller**
This adds the create/save/delete methods for invoiceTransactions.  I'm guessing there are a few items here I have missed, I would appreciate any feedback you guys have.  I have to add some docs but figured I'd make sure the code was correct first.

I'm also not sure on the npm publishing workflow for this, I also don't have access to that as well.

### CODE REVIEW

**Comment by Ollo** [comment]
@matthopson can you help us with the release / publishing process for this? Need this sdk method for Invoicing stories in this sprint. 🌮

**Comment by amberwebb** [comment]
@mikereinmiller Docs and changelog?

File: lib/teamsnap.js [hunk 1]
```diff
@@ -1650,6 +1650,33 @@ exports.loadInvoiceTransactions = function(params, callback) {
...
+  var ref;
+  if (!invoiceTransaction) {
+    throw new TSArgsError('teamsnap.saveInvoiceTransaction', "`invoiceTransaction` must be provided");
+  }
+  if (!this.isItem(invoiceTransaction, 'invoiceTransaction')) {
+    throw new TSArgsError('teamsnap.saveInvoiceTransaction', "`invoiceTransaction.type` must be 'invoiceTransaction'");
+  }
+  if (!invoiceTransaction.invoiceId) {
+    return this.reject('You must provide a invoice.', 'invoiceId', callback);
+  }
```
Comment (amberwebb) [inline lib/teamsnap.js]: `'You must provide an invoice.`

Comment (mikereinmiller) [inline lib/teamsnap.js]: good catch Amber

File: src/collections/invoiceTransactions.coffee [hunk 2]
```diff
@@ -4,3 +4,28 @@ exports.loadInvoiceTransactions = (params, callback) ->
...
+      "`invoiceTransaction.type` must be 'invoiceTransaction'"
+  unless invoiceTransaction.invoiceId
+    return @reject 'You must provide a invoice.', 'invoiceId', callback
+
+  @saveItem invoiceTransaction, callback
+
+
+exports.deleteInvoiceTransaction = (invoiceTransaction, callback) ->
+  unless invoiceTransaction
+    throw new TSArgsError 'teamsnap.deleteinvoiceTransaction',
```
Comment (amberwebb) [inline src/collections/invoiceTransactions.coffee]: Should this be `teamsnap.deleteInvoiceTransaction` ?

**Comment by amberwebb** [comment]
@mikereinmiller Need to update package.json.

File: docs/collections/invoice_transactions.md [hunk 3]
```diff
@@ -26,3 +28,86 @@ teamsnap.loadInvoiceTransactions({invoiceId: 1});
...
+```javascript
+// ~~~~~
+// Creates a new invoiceTransaction item.
+var invoiceTransaction = teamsnap.createInvoiceTransaction();
+
+// ~~~~~
+// Creates a new invoiceTransaction item with `invoiceId: 1`.
+var invoiceTransaction = teamsnap.createInvoiceTransaction({
+  invoiceId: 1,
+  ...additionalFields
```
Comment (matthopson) [inline docs/collections/invoice_transactions.md]: Are there additional required fields?

File: docs/collections/invoice_transactions.md [hunk 4]
```diff
@@ -26,3 +28,86 @@ teamsnap.loadInvoiceTransactions({invoiceId: 1});
...
+Deletes a `invoiceTransaction` item.
+
+### Params
+* `invoiceTransaction`: [object] - an item to be deleted.
+* `callback`: [function] - callback to be executed when the operation completes.
+
+### Examples
+```javascript
+// ~~~~~
+// Deletes a invoiceTransaction item.
```
Comment (matthopson) [inline docs/collections/invoice_transactions.md]: "an"

File: src/collections/invoiceTransactions.coffee [hunk 5]
```diff
@@ -4,3 +4,28 @@ exports.loadInvoiceTransactions = (params, callback) ->
...
+
+exports.saveInvoiceTransaction = (invoiceTransaction, callback) ->
+  unless invoiceTransaction
+    throw new TSArgsError 'teamsnap.saveInvoiceTransaction',
+      "`invoiceTransaction` must be provided"
+  unless @isItem invoiceTransaction, 'invoiceTransaction'
+    throw new TSArgsError 'teamsnap.saveInvoiceTransaction',
+      "`invoiceTransaction.type` must be 'invoiceTransaction'"
+  unless invoiceTransaction.invoiceId
+    return @reject 'You must provide a invoice.', 'invoiceId', callback
```
Comment (matthopson) [inline src/collections/invoiceTransactions.coffee]: "an"

File: docs/collections/invoice_transactions.md (hunk 4 above)
Comment (mikereinmiller) [inline docs/collections/invoice_transactions.md]: you guys and these Eagle "grammar" Eyes

File: docs/collections/invoice_transactions.md (hunk 3 above)
Comment (mikereinmiller) [inline docs/collections/invoice_transactions.md]: working with @rygr now, to figure out what they are exactly

**Comment by mikereinmiller** [comment]
@matthopson @amberwebb Could you guys take a look at this updates.  Added Documentation, Required Fields and removed delete method?

**Comment by matthopson** [comment]
@mikereinmiller Sorry, I could have been less lazy and actually looked for myself. For future reference, you can find required fields for endpoints via its affirmation params in v3: https://github.com/teamsnap/apiv3/blob/master/app/invoice_transactions/invoice_transaction_affirmation.rb

**Pull request merged**